### PR TITLE
fixed incorrect computation of symmetrical RCC3 relations, closes #40

### LINKED ISSUE #40: [rcc3] dc and po at the same time?
We (@pet1330 and I) just had a look at the rcc3 output of the example client:

```
Response is:
0: o2,o1:dc; o2,o3:dc; o1,o3:dc; o1,o2:dc; o3,o2:dc; o3,o1:po; 
1: o2,o1:dc; o2,o3:dc; o1,o3:dc; o1,o2:dc; o3,o2:dc; o3,o1:po; 
2: o2,o1:dc; o2,o3:dc; o1,o3:po; o1,o2:dc; o3,o2:dc; o3,o1:po;
```

How can o1,o3 be disconnected and o3,o1 be partially overlapping at the same time? Also the objects shouldn't overlap looking at the coordinates and bounding boxes. Something might be wrong with the logic here.

### PR DESCRIPTION
**Pull request by yianni**

### CODE REVIEW

**Comment by marc-hanheide** [comment]
@pet1330 can you have a look please?

**Comment by pet1330** [comment]
It's working correctly when using the example files, however, as soon as I added an additional object to the world I got the following error:
```
./example_standalone rcc3
Types of QSRs that have been included so far in the lib are the following:
- arg_relations_distance
- qtc_b_simplified
- qtc_bc_simplified
- qtc_c_simplified
- rcc3_rectangle_bounding_boxes_2d

Types of QSRs that you enabled are:
- arg_relations_distance
- qtc_b_simplified
- qtc_bc_simplified
- qtc_c_simplified
- rcc3_rectangle_bounding_boxes_2d

Traceback (most recent call last):
  File "./example_standalone", line 77, in <module>
    out = qsrlib.request_qsrs(request_message=request_message)
  File "/home/peter/workspace/src/strands_qsr_lib/qsr_lib/src/qsrlib/qsrlib.py", line 136, in request_qsrs
    qsr_relations_and_values=self.request_message.qsr_relations_and_values)
  File "/home/peter/workspace/src/strands_qsr_lib/qsr_lib/src/qsrlib_qsrs/qsr_abstractclass.py", line 38, in get
    return self.make(*args, **kwargs)
  File "/home/peter/workspace/src/strands_qsr_lib/qsr_lib/src/qsrlib_qsrs/qsr_rcc3_rectangle_bounding_boxes_2d.py", line 74, in make
    qsr = QSR(timestamp=timestamp, between=between, qsr=self.__compute_qsr(bb1, bb2))
  File "/home/peter/workspace/src/strands_qsr_lib/qsr_lib/src/qsrlib_qsrs/qsr_rcc3_rectangle_bounding_boxes_2d.py", line 105, in __compute_qsr
    occluded_points = self.__count_occluded_points(bb1, bb2)
  File "/home/peter/workspace/src/strands_qsr_lib/qsr_lib/src/qsrlib_qsrs/qsr_rcc3_rectangle_bounding_boxes_2d.py", line 126, in __count_occluded_points
    if self.is_point_in_rectangle(p, bb2):
AttributeError: 'QSR_RCC3_Rectangle_Bounding_Boxes_2D' object has no attribute 'is_point_in_rectangle'
```

This also happened when I used the example_ros_client

**Comment by yianni** [comment]
Post/Email me the complete file that fails please.

**Comment by yianni** [comment]
Or alternatively compare with the following (which works fine for me with the ROS example client)
```python
    if which_qsr_argv == "rcc3":
        o1 = [Object_State(name="o1", timestamp=0, x=1., y=1., width=5., length=8.),
              Object_State(name="o1", timestamp=1, x=1., y=2., width=5., length=8.),
              Object_State(name="o1", timestamp=2, x=1., y=3., width=5., length=8.)]

        o2 = [Object_State(name="o2", timestamp=0, x=11., y=1., width=5., length=8.),
              Object_State(name="o2", timestamp=1, x=11., y=2., width=5., length=8.),
              Object_State(name="o2", timestamp=2, x=11., y=3., width=5., length=8.),
              Object_State(name="o2", timestamp=3, x=11., y=4., width=5., length=8.)]

        o3 = [Object_State(name="o3", timestamp=0, x=1., y=11., width=5., length=8.),
              Object_State(name="o3", timestamp=1, x=2., y=11., width=5., length=8.),
              Object_State(name="o3", timestamp=2, x=3., y=11., width=5., length=8.)]

        o4 = [Object_State(name="o4", timestamp=0, x=51., y=11., width=5., length=8.),
              Object_State(name="o4", timestamp=1, x=52., y=11., width=5., length=8.),
              Object_State(name="o4", timestamp=2, x=53., y=11., width=5., length=8.)]


        world.add_object_state_series(o1)
        world.add_object_state_series(o2)
        world.add_object_state_series(o3)
        world.add_object_state_series(o4)
```

File: qsr_lib/src/qsrlib_qsrs/qsr_rcc3_rectangle_bounding_boxes_2d.py [hunk 1]
```diff
@@ -90,13 +90,83 @@ def __return_all_possible_combinations(self, objects_names):
...
+                        (bb1[2], bb1[1]),
+                        (bb1[2], bb1[3]),
+                        (bb1[0], bb1[3]))
+        bb2_4corners = ((bb2[0], bb2[1]),
+                        (bb2[2], bb2[1]),
+                        (bb2[2], bb2[3]),
+                        (bb2[0], bb2[3]))
+
+        for p in bb1_4corners:
+            if self.is_point_in_rectangle(p, bb2):
```
Comment (pet1330) [inline qsr_lib/src/qsrlib_qsrs/qsr_rcc3_rectangle_bounding_boxes_2d.py]: This is missing the two leading underscores. This is only called if you have overlapping objects. By adding the two underscores it seems to work

Comment (yianni) [inline qsr_lib/src/qsrlib_qsrs/qsr_rcc3_rectangle_bounding_boxes_2d.py]: Good catch. :+1: Missed out from the debugging file. Now, fixed and committed.

**Comment by cdondrup** [comment]
retest this please

**Comment by cdondrup** [comment]
Should be fine, @pet1330 agrees. Will merge.

**Pull request merged**